frontend/src/libs/hooks/use-pagination/use-pagination.hook.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useRef, useState } from 'react';
+
+import {
+  DEFAULT_PAGINATION_SKIP,
+  DEFAULT_PAGINATION_TAKE,
+} from './libs/constants/constants.js';
+import {
+  type UsePaginationProperties,
+  type UsePaginationReturn,
+} from './libs/types/types.js';
+
+const usePagination = ({
+  defaultSkip = DEFAULT_PAGINATION_SKIP,
+  defaultTake = DEFAULT_PAGINATION_TAKE,
+}: UsePaginationProperties = {}): UsePaginationReturn => {
+  const [hasMore, setHasMore] = useState(false);
+  const paginationParameters = useRef({ skip: defaultSkip, take: defaultTake });
+
+  const load = useCallback(
```

Comment:
Let's call it `loadMore` - every time I open diff, it brings me a thought that in components `load` (from http-api) is called directly
```suggestion
  const loadMore = useCallback(
```